web/src/stream_events.js

```diff
@@ -199,10 +200,16 @@ export function remove_deactivated_user_from_all_streams(user_id) {
     }
 }
 
-export function process_subscriber_update(stream_ids) {
+export function process_subscriber_update(user_ids, stream_ids) {
     for (const stream_id of stream_ids) {
         const sub = sub_store.get(stream_id);
         stream_settings_ui.update_subscribers_ui(sub);
     }
     compose_fade.update_faded_users();
+    if (overlays.get_user_id_if_user_profile_modal_open()) {
+        const user_id = overlays.get_user_id_if_user_profile_modal_open();
+        if (user_ids.includes(user_id)) {
+            user_profile.update_user_profile_streams_list(user_id);
+        }
+    }
```

Comment:
I think this is a weird abstraction boundary; can we have this function call `update_profile.update_user_profile_streams_list_for_user(user_id)` and have that function contain this the logic to return if there's nothing to do? It makes much more sense to me for the `user_profile` module to be the one responsible for deciding how to handle the event to updates its UI with that logic all in one place.

Reply:
Done, do let me know if you want me to make more change.

Reply:
This needs more work -- I want this function to just call `user_profile.update_user_profile_streams_list_for_users(user_ids)`, and have all the logic that you currently have both here and `get_user_id_if_user_profile_modal_open` be in that one place, rather than split between three places in modules that are not responsible for the user profile modal.

Reply:
I think this will also eliminate the need for the second commit entirely -- we can just check what modal is open directly inside `user_profile.js`.

Reply:
Thanks for the suggestion, for the second commit I have replaced that with the implementation of the same function but inside the `user_profile.js`, and refactored the functions you mentioned above.